core/src/impl/Kokkos_ClockTic.hpp

```diff
@@ -52,6 +52,12 @@
 #include <omp.h>
 #endif
 
+#if defined(KOKKOS_ENABLE_SYCL) && defined(KOKKOS_ARCH_INTEL_GEN) && \
+    defined(__SYCL_DEVICE_ONLY__)
+extern SYCL_EXTERNAL unsigned long __attribute__((overloadable))
+intel_get_cycle_counter();
```

Comment:
Comment that to use OpenCL(TM) built-In intrinsics inside kernels we have to declare their prototype.